packages/firestore/src/local/shared_client_state.ts

```diff
@@ -473,18 +477,10 @@ export class LocalClientState implements ClientState {
   }
 
   addQueryTarget(targetId: TargetId): void {
-    assert(
-      !this.activeTargetIds.has(targetId),
-      `Target with ID '${targetId}' already active.`
-    );
```

Comment:
Hrm. Why exactly did these asserts go away?

Reply:
I added the first one back. I removed both of them to avoid the question why only this method has an assert.

 `removeQueryTarget` can't have an assert anymore since we call this method twice during  `SyncEngine.unlisten`. Once in the beginning to compute whether the target will remain active, and once when we call `SyncEngine.removeAndCleanUpQueryData`. We can potentially rework the `targetRemainsActive` logic (something along the lines of `SharedClientState.amITheOnlyClient(...)`) and add this assert back.